src/go/rdctl/pkg/factoryreset/delete_data.go

```diff
@@ -0,0 +1,307 @@
+}
+
+func getStandardDirs() (string, string, string, error) {
+	configDir, err := os.UserConfigDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	return configDir, cacheDir, homeDir, nil
+}
+
+func deleteDarwinData(removeKubernetesCache bool) error {
+	configDir, cacheDir, homeDir, err := getStandardDirs()
+	if err != nil {
+		return err
+	}
+	libraryPath := path.Join(homeDir, "Library")
+
+	altAppHomePath := path.Join(homeDir, ".rd")
+	appHomePath := path.Join(configDir, "rancher-desktop")
+	cachePath := path.Join(cacheDir, "rancher-desktop")
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath == "" {
+		logsPath = path.Join(libraryPath, "Logs", "rancher-desktop")
+	}
+	settingsPath := path.Join(libraryPath, "Preferences", "rancher-desktop")
+	updaterPath := path.Join(configDir, "Caches", "rancher-desktop-updater")
+
+	pathList := []string{
+		altAppHomePath,
+		appHomePath,
+		logsPath,
+		settingsPath,
+		updaterPath,
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteUnixLikeData(homeDir, altAppHomePath, path.Join(homeDir, ".config"), pathList)
+}
+
+func deleteUnixData(removeKubernetesCache bool) error {
+	configHomePath, cacheHomePath, homeDir, err := getStandardDirs()
+	if err != nil {
+		return err
+	}
+
+	dataDir := os.Getenv("XDG_DATA_HOME")
+	if dataDir == "" {
+		dataDir = path.Join(homeDir, ".local", "share")
+	}
+	altAppHomePath := path.Join(homeDir, ".rd")
+	cachePath := path.Join(cacheHomePath, "rancher-desktop")
+	configPath := path.Join(configHomePath, "rancher-desktop")
+	dataHomePath := path.Join(dataDir, "rancher-desktop")
+
+	pathList := []string{
+		altAppHomePath,
+		configPath,
+		path.Join(homeDir, ".local", "state", "rancher-desktop"),
+	}
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath != "" {
+		pathList = append(pathList, logsPath, path.Join(dataHomePath, "lima"))
+	} else {
+		pathList = append(pathList, path.Join(dataHomePath))
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteUnixLikeData(homeDir, altAppHomePath, configHomePath, pathList)
+}
+
+func unregisterAndDeleteWindowsData(removeKubernetesCache bool) error {
+	if err := unregisterWSL(); err != nil {
+		return err
+	}
+	if err := deleteWindowsData(!removeKubernetesCache, "rancher-desktop"); err != nil {
+		return err
+	}
+	return clearDockerContext()
+}
+
+// Most of the errors in this function are reported, but we continue to try to delete things,
+// because there isn't really a dependency graph here.
+// For example, if we can't delete the Lima VM, that doesn't mean we can't remove docker files
+// or pull the path settings out of the shell profile files.
+func deleteUnixLikeData(homeDir string, altAppHomePath string, configHomePath string, pathList []string) error {
```

Comment:
This would be more properly `deleteUnixData` (going by golang predefined build tags).